example/src/main/java/io/realm/examples/adapters/ui/recyclerview/RecyclerViewExampleActivity.java

```diff
@@ -66,32 +68,81 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         int id = item.getItemId();
-        if (id == R.id.action_add) {
-            final String timestamp = Long.toString(System.currentTimeMillis());
-            realm.executeTransactionAsync(new Realm.Transaction() {
-                @Override
-                public void execute(Realm realm) {
-                    realm.createObject(TimeStamp.class).setTimeStamp(timestamp);
-                }
-            });
-            return true;
+        switch(id) {
+            case R.id.action_add:
+                addItem();
+                return true;
+            case R.id.action_random:
+                randomEditItem();
+                return true;
+            case R.id.action_delete_all:
+                deleteAllItems();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
         }
-        return super.onOptionsItemSelected(item);
     }
 
     private void setUpRecyclerView() {
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
-        recyclerView.setAdapter(new MyRecyclerViewAdapter(this, realm.where(TimeStamp.class).findAllAsync()));
+        recyclerView.setAdapter(new MyRecyclerViewAdapter(
+                this, realm.where(Counter.class).findAllSortedAsync(Counter.FIELD_COUNT)));
         recyclerView.setHasFixedSize(true);
         recyclerView.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL_LIST));
     }
 
-    public void deleteItem(TimeStamp item) {
-        final String id = item.getTimeStamp();
+    // Randomly duplicate/delete/create some objects in the Realm.
+    private void randomEditItem() {
+        realm.executeTransactionAsync(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                Random rand = new Random();
```

Comment:
I think `Random` is a bit expensive to init. better create it as a member I guess.